packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_stats_table.dart

```diff
@@ -124,11 +125,14 @@ class ClassStatsTable extends StatefulWidget {
 class _ClassStatsTableState extends State<ClassStatsTable>
     with AutoDisposeMixin {
   late final List<ColumnData<_RetainingPathRecord>> _columns;
+  late final List<MapEntry<ClassOnlyHeapPath, SizeOfClassSet>> _dataList;
 
   @override
   void initState() {
     super.initState();
 
+    _dataList = widget.data.sizeByRetainingPath.entries.toList(growable: false);
```

Comment:
_dataList needs to be re-initialized in didUpdate widget when widget.data has changed

Reply:
fixed, but i am not sure i understand it

Reply:
My comment was for `didUpdateWidget`, not `didChangeDependencies`. For a stateful widget, `didUpdateWidget` is called when the widget is rebuilt, so if the widget was rebuilt with a new value for `data`, the state needs to be updated to reflect the changed value. A good rule of thumb is any time you are doing some sort of data initialization in `initState` or `didChangeDependencies` based on the widget's values, the same initialization will need to take place in `didUpdateWidget` to ensure the state is updated properly when the widget changes.